plugins/woocommerce-blocks/tests/e2e/tests/product-collection/product-collection.page.ts

```diff
@@ -455,26 +459,52 @@ class ProductCollectionPage {
 		return await orderByComboBox.inputValue();
 	}
 
+	getOnSaleControl() {
+		const sidebarSettings = this.locateSidebarSettings();
+		return sidebarSettings.getByRole( 'radiogroup', {
+			name: SELECTORS.onSaleControlLabel,
+		} );
+	}
+
+	getOnSaleControlShowOnlyOption() {
+		const onSaleControl = this.getOnSaleControl();
+		return onSaleControl.getByRole( 'radio', {
+			name: SELECTORS.onSaleShowOnlyControlOptionLabel,
+		} );
+	}
+
+	getOnSaleControlDontShowOption() {
+		const onSaleControl = this.getOnSaleControl();
+		return onSaleControl.getByRole( 'radio', {
+			name: SELECTORS.onSaleDontShowControlOptionLabel,
+		} );
+	}
```

Comment:
I’d love to move away from using the `SELECTORS` object convention altogether. This approach stems from a time when our only options for selectors were class names, IDs, or attributes. However, with Playwright’s extensive Locator API, maintaining this convention has become more of a burden for a couple of reasons:

- We need to consciously remember to extract selectors to maintain consistency.
- It’s challenging to determine what should be extracted, as our mindset has shifted from thinking in selectors to thinking in locators.
- It reduces the readability of tests, as it’s not immediately obvious which element we are interacting with.

What do you think, @imanish003 @kmanijak @mdperez86?

To be clear, this is not a request to change anything here - just a conversation starter. I'll create an issue if we agree on this point.

Reply:
It is a good point and it sounds good to me!

Reply:
@WunderBart 

> I’d love to move away from using the SELECTORS object convention altogether.

Sounds good to me. Just to confirm:
- Are you suggesting:
	- This: `locator('.wc-block-product-template')`
	- Instead of: `locator(SELECTORS.productTemplate)`
- OR, Are you suggesting that we should stop depending on class names, IDs, or attributes altogether? In other words, we should only use [Locator methods](https://playwright.dev/docs/locators) like:
	- `page.getByRole()`
	- `page.getByText()`
	- `page.getByLabel()` etc?

I feel like you are suggesting 2nd option, but just want to confirm it 🙂

Reply:
Great questions! The [general recommendation](https://playwright.dev/docs/best-practices#prefer-user-facing-attributes-to-xpath-or-css-selectors) is to use user-facing attributes as much as possible for stability and readability. Although class names are considered API in WordPress (meaning they aren't as likely to change, so the stability might not be an issue), it's still better to use user-facing attributes as they're closer to the actual user interaction and make the tests easier to follow.

To answer your questions directly:

> Are you suggesting:
> - This: locator('.wc-block-product-template')
> - Instead of: locator(SELECTORS.productTemplate)

In cases where we're forced to use a CSS selector, let's use the string directly (e.g. `page.locator( '.wp-button' )`). It's not an issue if we're repeating the string — we can always mass-rename if necessary. It's also at least one less thing to worry about – naming the constants for the selectors (and we know how hard naming is! 😅)

> OR, Are you suggesting that we should stop depending on class names, IDs, or attributes altogether? In other words, we should only use [Locator methods](https://playwright.dev/docs/locators) like:
> - page.getByRole()
> - page.getByText()
> - page.getByLabel() etc?

I don't think we'll ever be able to stop depending entirely on CSS selectors, so I'm not suggesting that. Wherever possible, I recommend preferring the user-facing attributes handled mainly by the `getBy*` locators.

Does that make sense?